hail/src/main/scala/is/hail/methods/VEP.scala

```diff
@@ -96,12 +98,25 @@ object VEP {
       None
     }
   }
+}
+
+case class VEP(config: String, csq: Boolean, blockSize: Int) extends TableToTableFunction {
+  private val conf = VEP.readConfiguration(HailContext.get.hadoopConf, config)
+  private val vepSignature = conf.vep_json_schema
+
+  override def preservesPartitionCounts: Boolean = false
 
-  def annotate(ht: Table, config: String, csq: Boolean, blockSize: Int): Table = {
-    assert(ht.key == FastIndexedSeq("locus", "alleles"))
-    assert(ht.typ.rowType.size == 2)
+  override def typeInfo(childType: TableType, childRVDType: RVDType): (TableType, RVDType) = {
+    val vepType = if (csq) TArray(TString()) else vepSignature
+    val t = TableType(childType.rowType ++ TStruct("vep" -> vepType), FastIndexedSeq("locus", "alleles"), childType.globalType)
```

Comment:
feels like we should also `assert tv.typ.key == FastIndexedSeq("locus", "alleles")` or use `childType.key` here.